library/src/main/java/io/ona/kujaku/downloaders/MapBoxOfflineResourcesDownloader.java

```diff
@@ -207,7 +208,8 @@ private void checkDownloadMapParams(String name, String styleUrl, double minZoom
             throw new OfflineMapDownloadException("Invalid map name");
         }
 
-        if (TextUtils.isEmpty(styleUrl)) {
+        if (TextUtils.isEmpty(styleUrl) || !(styleUrl.matches(Constants.MAP_BOX_URL_FORMAT) ||
+                styleUrl.startsWith("https://") || styleUrl.startsWith("http://"))) {
```

Comment:
Won't ` styleUrl.startsWith("https://") || styleUrl.startsWith("http://")` be caught by
`!(styleUrl.matches(Constants.MAP_BOX_URL_FORMAT)`?

Reply:
Nope, it won't. The url formats are different

Reply:
You can try this code to test this: 

```


		
String styleUrl = "mapbox://styles/mapbox/light";
String MAP_BOX_URL_FORMAT = "mapbox://styles/[A-Za-z0-9]+/[A-Za-z0-9]+";
if (!(styleUrl.matches(MAP_BOX_URL_FORMAT) ||
                styleUrl.startsWith("https://") || styleUrl.startsWith("http://"))) {
            System.out.println("Hello here");
}
```

Reply:
Remove the last two conditionals and add `http://` or `https://` to the styleUrl

Reply:
The aim is to allow `mapbox://styles/[A-Za-z0-9]+/[A-Za-z0-9]` and any `https` and `http` url which links to a valid style and not any other string. In a case where we have Reveal as it is having a hosted style that is accessible currently only through an actual `HTTP/HTTPS` url, Mapbox offline download manager allows that as a valid style URL and we should too otherwise Reveal and other apps will not be able to make offline downloads of such styles and their resources/tiles just because we introduced a bug.

Reply:
Cool, got it.